workflows/short-read-mngs/host_filter.wdl

```diff
@@ -232,6 +245,94 @@ task RunValidateInput {
+  input {
+    File valid_input1_fastq
+    File? valid_input2_fastq
+
+    # GENOME_NAME.bowtie2.tar should contain GENOME_NAME/GENOME_NAME.*.bt*
+    File ercc_index_tar
+    String bowtie2_options = "--very-sensitive-local"
+
+    String docker_image_id
+    Int cpu = 16
+  }
+
+  Boolean paired = defined(valid_input2_fastq)
+  String genome_name = "ercc" 
+  String bowtie2_invocation =
+      "bowtie2 -x '/tmp/${genome_name}/${genome_name}' ${bowtie2_options} -p ${cpu}"
+        + (if (paired) then " -1 '${valid_input1_fastq}' -2 '${valid_input1_fastq}'" else " -U '${valid_input1_fastq}'")
+        + " -q -S '/tmp/bowtie2_ercc.sam'"
+
+  command <<<
+    set -euxo pipefail
+
+    tar xf '~{ercc_index_tar}' -C /tmp
+
+    ~{bowtie2_invocation}
+
+    # Extract reads [pairs] that did NOT map to the index
+    if [[ '~{paired}' == 'true' ]]; then
+        #    1 (read paired)
+        #    4 (read unmapped)
+        # +  8 (mate unmapped)
+        # ----
+        #   13
+        samtools fastq -f 13 -1 'bowtie2_ercc_filtered1.fastq' -2 'bowtie2_ercc_filtered2.fastq' -0 /dev/null -s /dev/null /tmp/bowtie2_ercc.sam
+        count="$(cat bowtie2_ercc_filtered{1,2}.fastq | wc -l)"
+    else
+        samtools fastq -f 4 /tmp/bowtie2_ercc.sam > 'bowtie2_ercc_filtered1.fastq'
+        count="$(cat bowtie2_ercc_filtered1.fastq | wc -l)"
+    fi
+
+    
+    # Calculate ercc counts for bowtie2
+    samtools view /tmp/bowtie2_ercc.sam | cut -f3 |  (grep "ERCC-" || [ "$?" == "1" ])| sort | uniq -c | awk '{ print $2 "\t" $1}' > 'bowtie2_ERCC_counts.tsv'
+
+    count=$((count / 4))
+    jq --null-input --arg count "$count" '{"bowtie2_ercc_filtered_out":$count}' > 'bowtie2_ercc_filtered_out.count'
+
+    python3 - << 'EOF'
+    import textwrap
+    with open("bowtie2.description.md", "w") as outfile:
+      print(textwrap.dedent("""
+      **bowtie2 ercc filtering**
+
+      Filters out reads matching the ercc genome using
+      [Bowtie2](https://bowtie-bio.sourceforge.net/bowtie2/index.shtml). Runs
+      `bowtie2 ~{bowtie2_options}` using a precomputed index, then uses
+      [samtools](http://www.htslib.org/) to keep reads *not* mapping to the ercc genome.
+
+      Bowtie2 is run on the fastp-filtered FASTQ(s):
+
+      ```
+      ~{bowtie2_invocation}
+      ```
+
+      Then, non-mapping reads are selected using `samtools fastq -f ~{if (paired) then 13 else 4}`.
+
+      Bowtie2 documentation can be found [here](https://bowtie-bio.sourceforge.net/bowtie2/index.shtml)
+      """).strip(), file=outfile)
+    EOF
+
+  >>>
+
+  output {
+    String step_description_md = read_string("bowtie2.description.md")
+    File bowtie2_ercc_filtered1_fastq = "bowtie2_ercc_filtered1.fastq"
+    File bowtie2_ercc_counts = "bowtie2_ERCC_counts.tsv"
+    File? bowtie2_ercc_filtered2_fastq = "bowtie2_ercc_filtered2.fastq"
+    File reads_out_count = "bowtie2_ercc_filtered_out.count"
+  }
+  runtime {
+    docker: docker_image_id
+    cpu: cpu
+    memory: "1G"
```

Comment:
Is this the right setting of `memory`? The memory usage should be low with only the small ERCC sequences in the bowtie2 index, but 1G might be cutting it a little close